### PR TITLE
Added Speed Property to mediaelement

### LINKED ISSUE #649: [Enhancement] MediaElement - Add Playback Speed Controls
## Summary

Please add playback speed controls as optional on the toolbar for every platform

## API Changes

.ShowPlaybackSpeed = true/false

## Intended Use Case
Anyone using the video player would find this valuable.

## Who Will Do The Work?

- [ ] I am willing to take this on myself
- [X] Just putting this out there for someone to pick up

### PR DESCRIPTION
**Pull request by AswinPG**
### Description of Change ###

Added Speed property for Mediaelement in iOS and Android

### Bugs Fixed ###

- Fixes #649

### PR Checklist ###

- [x] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by jfversluis** [comment]
Seems to work great on Android and iOS at least, thanks for this!

Would you be able to add this to the sample app too? Ideally some way to change the new `Speed` property in the `MediaElement` page? A slider maybe? Thanks!

**Comment by AswinPG** [comment]
@jfversluis I have added the sample in the media element page :)

**Comment by jfversluis** [comment]
One last request: could you rebase it please? Or at least update it with the latest changes? I would do it for you, but I don't have the rights to your branch :)

**Comment by AswinPG** [comment]
> One last request: could you rebase it please? Or at least update it with the latest changes? I would do it for you, but I don't have the rights to your branch :)
Did it :)

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/MediaElementRenderer.android.cs
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/MediaElementRenderer.android.cs:387]:
```suggestion
```
Is it needed?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/MediaElementRenderer.android.cs
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/MediaElementRenderer.android.cs:386]:
```suggestion
			if (MediaElement == null || mediaPlayer == null)
				return;

			var playbackParams = new PlaybackParams();
			playbackParams.SetSpeed((float)MediaElement.Speed);
			mediaPlayer.PlaybackParams = playbackParams;
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs:194]: What's the purpose of adding this if this property is implemented implicitly? (And does the same)

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs:51]: Should it be OneWay?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/MediaElementRenderer.android.cs
Comment (AswinPG) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/MediaElementRenderer.android.cs:387]: Sorry I was in a hurry and did not see that

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs
Comment (AswinPG) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs]: Yes.. I've made it oneway now

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs
Comment (AswinPG) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/MediaElement.shared.cs]: Removed that too 👍

**Comment by jfversluis** [comment]
OK, so sorry. One last rebase and then it should merge automatically... 😇

**Review by thanhxuann-clearwave — APPROVED** [review]
LGTM

**Comment by jfversluis** [comment]
> OK, so sorry. One last rebase and then it should merge automatically... 😇

OMG... WHY DIDN'T IT DO IT?!

**Review by jfversluis — APPROVED** [review]

**Comment by AswinPG** [comment]
> > OK, so sorry. One last rebase and then it should merge automatically... 😇
> 
> OMG... WHY DIDN'T IT DO IT?!

Is there something that I should do ?

**Comment by pictos** [comment]
@AswinPG looks like this branch is out-of-date, can you rebase your branch against develop and push?

**Comment by AswinPG** [comment]
> @AswinPG looks like this branch is out-of-date, can you rebase your branch against develop and push?

It's my first time rebasing. I don't really know if I did it right or wrong...

**Comment by pictos** [comment]
@AswinPG yeah, something went wrong... We can see based on the number of changed files

**Comment by pictos** [comment]
I see... Looks like that you made your changes against our `main` branch instead of the `develop` branch.

**Comment by pictos** [comment]
@AswinPG looks like everything is ok now, let's how it goes. I'll let @jfversluis take a final look at this since this rebase dismissed his review.

**Comment by AswinPG** [comment]
> @AswinPG looks like everything is ok now, let's how it goes. I'll let @jfversluis take a final look at this since this rebase dismissed his review.

Thanks @pictos

**Comment by AswinPG** [comment]
> @AswinPG looks like everything is ok now, let's how it goes. I'll let @jfversluis take a final look at this since this rebase dismissed his review.

Thanks @pictos

**Comment by jfversluis** [comment]
Implemented through #1570 to work around asking for another rebase :)

Thanks so much @AswinPG !

**Comment by AswinPG** [comment]
Thanks @jfversluis :)

**Pull request closed without merging**
